src/data/mod.rs

```diff
@@ -42,6 +42,10 @@
 mod macros;
 
 use crate::shared::*;
+
+#[cfg(feature = "ahash")]
+use ahash::HashMap;
```

Comment:
Also, explain me please about this thing. What it should be here?

Reply:
ahash is a hashmap library with faster hashing. I refactored it since then to use a boxed slice instead, which should be even faster for what we're doing.

Reply:
[Here's](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=use+ahash%3A%3AAHashMap%3B%0Ause+std%3A%3Acollections%3A%3AHashMap%3B%0Ause+std%3A%3Atime%3A%3A%7BInstant%2C+Duration%7D%3B%0A%0Aconst+LOREM%3A+%26str+%3D+%22Lorem+ipsum+dolor+sit+amet%2C+consectetur+adipiscing+elit%2C+sed+do+eiusmod+tempor+incididunt+ut+labore+et+dolore+magna+aliqua.+Ut+enim+ad+minim+veniam%2C+quis+nostrud+exercitation+ullamco+laboris+nisi+ut+aliquip+ex+ea+commodo+consequat.+Duis+aute+irure+dolor+in+reprehenderit+in+voluptate+velit+esse+cillum+dolore+eu+fugiat+nulla+pariatur.+Excepteur+sint+occaecat+cupidatat+non+proident%2C+sunt+in+culpa+qui+officia+deserunt+mollit+anim+id+est+laborum.%22%3B%0A%0Afn+main%28%29+%7B%0A++++let+lorem_ahashmap%3A+AHashMap%3C_%2C+_%3E+%3D+LOREM.split%28%27+%27%29.enumerate%28%29.collect%28%29%3B%0A++++let+lorem_std_hashmap%3A+HashMap%3C_%2C+_%3E+%3D+LOREM.split%28%27+%27%29.enumerate%28%29.collect%28%29%3B%0A++++let+lorem_vec%3A+Vec%3C%28_%2C+_%29%3E+%3D+LOREM.split%28%27+%27%29.enumerate%28%29.collect%28%29%3B%0A++++%2F%2F+then+test+lorem_boxed%2C+changing+the+type+sig+in+the+function+to+%0A++++%2F%2F+let+lorem_boxed%3A+Box%3C%5B%28_%2C+_%29%5D%3E+%3D+LOREM.split%28%27+%27%29.enumerate%28%29.collect%28%29%3B%0A++++%2F%2F+I+can%27t+test+it+as+an+array%2C+the+Boxed+example+is+probably+the+closest.%0A++++%0A++++iter_test%28%22lorem_ahashmap%22%2C+lorem_ahashmap%29%3B%0A++++iter_test%28%22lorem_std_hashmap%22%2C+lorem_std_hashmap%29%3B%0A++++iter_test%28%22lorem_vec%22%2C+lorem_vec%29%3B%0A++++%2F%2F+iter_test%28%22lorem_boxed%22%2C+lorem_boxed%29%3B%0A%7D%0Afn+iter_test%3CI%3A+IntoIterator%3CItem+%3D+%28usize%2C+%26%27static+str%29%3E%3E%28test_name%3A+%26str%2C+iter%3A+I%29+%7B%0A++++let+inst+%3D+std%3A%3Atime%3A%3AInstant%3A%3Anow%28%29%3B%0A++++let+num+%3D+iter.into_iter%28%29.map%28%7C%28k%2C+v%29%7C+k+%2B+1%29.collect%3A%3A%3CVec%3C_%3E%3E%28%29.len%28%29%3B%0A++++let+since+%3D+inst.elapsed%28%29%3B%0A%0A++++println%21%28%22Test+%7Btest_name%7D+took+%7B%7Dns%22%2C+since.as_nanos%28%29%29%3B%0A%7D) a rust playground link to some benchmarks. I found that just changing it to a vec resulted in double the performance. I couldn't figure out lifetimes in the playground editor to get the Boxed slice working, and I didn't do arrays. However, these are theoretically faster than Vecs. I changed the implementation to an array, because we don't need anything all that fancy.